controllers/controller.go

```diff
@@ -209,7 +209,13 @@ func (i instanceReconcilerHelper) reconcileInstance(ctx context.Context, o clien
 
 	if !isAlreadyProcessed(o) {
 		i.rec.Event(o, corev1.EventTypeNormal, eventCreateOrUpdatedAtAiven, "about to create instance at aiven")
-		if err := i.createOrUpdateInstance(o, refs); err != nil {
+		err := i.createOrUpdateInstance(o, refs)
```

Comment:
As mentioned before, we can avoid this additional error handling here if we move the `read_replica` service integration check to the `checkPreconditions` method of the service. 

We already have logic in our controller handler responsible for checking all possible dependencies we need before we start the creation or update of the service.